pkg/server/cache/entrycache/fullcache.go

```diff
@@ -173,6 +179,37 @@ func Build(ctx context.Context, entryIter EntryIterator, agentIter AgentIterator
 	}, nil
 }
 
+func Update(ctx context.Context, ds datastore.DataStore, cache *FullEntryCache) error {
+	req := &datastore.ListRegistrationEntriesEventsRequest{
+		LastID: cache.lastEventID,
+	}
+	resp, err := ds.ListRegistrationEntriesEvents(ctx, req)
```

Comment:
I'm wondering if it would make more sense for `ListRegistrationEntries` to get supported for fetching entries based on the events table, similarly to how it can list them by selectors or federation relationships.

This would avoid fetching a potentially long list of change events and then fetching each entry individually which would involve multiple round trips to the database.

Reply:
That's what i was hoping to do too. Unfortunately that won't capture delete events.